### PR TITLE
Fix drop command, implement data_requests table, implement aggregation CSV file downloads, remove user_energy function

### PR DESCRIPTION
**Pull request by wheeendiii**
# Description

- Drop command bug fix. #7 
- Data_request table check/setup in the main loop and read/direct in all unprocessed rows to respective functions.
- CSV file download to all aggregation functions and add file location to data_request table.
- User_energy remove temporarily. #8 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking chan
Fixes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Testing 
- [ ] Change that requires a documentation update

# How Has This Been Tested?

Run query_test.py file to check output CSV to the current folder.
Run unit test query_test.py and check the result of unit tests (should say "ok").
Manually test edge cases like null inputs and an empty table. (output error messages)

(to run unit test: cd komodo/komodo-data and run: python -m unittest query_test.py           

- [x] Manual Test
- [x] Unit Test
- [ ] Integration / End-to-End Test

**Test Configuration**:

* Python version 3.8.11
* Mysql 8

# Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas that are not self-documenting
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings 
- [x] My changes have no unnecessary logging
- [x] I have added tests that prove my fix is effective or that my feature works, for sufficiently complex features
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] Sensitive info like tokens, secrets, and passwords have been removed before submitting

Modified from this article:
Phillip Johnston, “A GitHub Pull Request Template for Your Projects - Embedded Artistry,” Embedded Artistry, Aug. 04, 2017. https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ (accessed Jul. 22, 2021).

### CODE REVIEW

File: process.py [hunk 1]
```diff
@@ -48,21 +49,23 @@ def check_for_unprocessed_captures():
 
     return ready
 
-def aggregate_interaction_type(session_id, interaction_type):
+def aggregate_interaction_type(session_id, interaction_type,request_id):
```
Comment (parseccentric) [inline process.py]: add space between args

File: process.py [hunk 2]
```diff
@@ -81,27 +84,44 @@ def aggregate_interaction_type(session_id, interaction_type):
...
 
-        
-    return True
+            with conn.begin():
+                query = text("""
+                SELECT * 
+                FROM aggregate_interaction;
+                """
+                )
+                result = conn.execute(query)
```
Comment (parseccentric) [inline process.py:94]: try to add empty lines for clarity where needed

File: process.py [hunk 3]
```diff
@@ -81,27 +84,44 @@ def aggregate_interaction_type(session_id, interaction_type):
...
+                update_data_request(request_id, 1, file_path)
+                return True
 
-def aggregate_user(session_id,client_id):
+    except Exception as e:
+        print(e)
+        return False
+    
+
+def aggregate_user(session_id,client_id,request_id):
```
Comment (parseccentric) [inline process.py]: documentation: explain return values in a comment

File: process.py (hunk 1 above)
Comment (parseccentric) [reply process.py]: documentation: explain return values in a comment

File: process.py [hunk 4]
```diff
@@ -202,7 +200,109 @@ def mark_as_processed(capture_id, success):
...
+                    primary key (request_id)
+                    );
+                    """
+                    )
+
+                    conn.execute(query)
+
+                with conn.begin(): 
+                    query = text("""
+                    INSERT INTO data_requests (`processed_capture_id`, `who_requested`, `aggregation_function`, `is_it_fulfilled`,`message`)
```
Comment (parseccentric) [inline process.py:251]: Option 1: Leave it here if needed. Change all values to be obviously dummy values (9999, dummy_function, etc.). Make sure the capture is marked as processed.

Option 2: Move this dummy insertion and check into the unit tests. (Leave values as-is if it's helpful)

File: process.py [hunk 5]
```diff
@@ -221,6 +321,11 @@ def mark_as_processed(capture_id, success):
             print('Nothing to process', time.strftime("%H:%M:%S", time.localtime()))
             # rinse & repeat
             time.sleep(10)
+        
+        # check data_request table and direct to respective functions
+        if check_for_data_requests_table():
```
Comment (parseccentric) [inline process.py:346]: (Future optimization) Move this to the outside of the loop, cache the result of this value in a boolean, and check the boolean (inside the loop).

File: query_test.py [hunk 6]
```diff
@@ -3,13 +3,17 @@
 from unittest import TestCase
 import numpy as np
 from numpy.testing._private.utils import assert_equal
+from sqlalchemy.sql.sqltypes import BIGINT, JSON, Integer
 import process 
-from sqlalchemy import create_engine, text
+from sqlalchemy import create_engine, text, over,Table, Column, String, MetaData
```
Comment (parseccentric) [inline query_test.py]: comma space :)

File: query_test.py [hunk 7]
```diff
@@ -46,10 +51,11 @@ def test_query(self):
                 """
                 )
                 result = conn.execute(query)
-                count = [r[1] for r in result]
+                count = [r[1:] for r in result]
                 sum = np.sum(count)
 
-        user_flag = process.aggregate_user(126,5) 
+        # aggregate_uer testing
```
Comment (parseccentric) [inline query_test.py]: typo

File: query_test.py [hunk 8]
```diff
@@ -60,34 +66,12 @@ def test_query(self):
...
 
-            result = conn.execute(query)
-            count = [r[0:] for r in result]
-            df = pd.DataFrame(count, columns = ['session_id','timestamp','entity_type','energy','energy_rank'])
-            df.to_csv('energy_out.csv',index=False)
-            print(df.head(15))
+        # aggregate_interaction
+        self.assertEqual(sum, 1288) 
+        # aggregate_user
+        self.assertEqual(sum2, 16797)
```
Comment (parseccentric) [inline query_test.py]: (Future cleanup) Try to test only one thing per `def <test_name>`. As a result, make sure that the setup is done in the beginning of each test or with python unittest syntactic sugar if possible.

**Review by parseccentric — CHANGES_REQUESTED** [review]
Please make changes not marked "(future xx)" and then I will approve! Thx

**Review by parseccentric — APPROVED** [review]
